plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -64,15 +63,8 @@ func (*AzureDataExplorer) SampleConfig() string {
 
 // Initialize the client and the ingestor
 func (adx *AzureDataExplorer) Connect() error {
-	authorizer, err := auth.NewAuthorizerFromEnvironmentWithResource(adx.Endpoint)
-	if err != nil {
-		return err
-	}
-	authorization := kusto.Authorization{
-		Authorizer: authorizer,
-	}
-	client, err := kusto.New(adx.Endpoint, authorization)
-
+	conn := kusto.NewConnectionStringBuilder(adx.Endpoint)
+	client, err := kusto.New(conn)
```

Comment:
In my version, #13524, I also used `WithDefaultAzureCredential` I am still not certain that this checks all the stated places for credentials though.

My concern is that the output specifies a number of places that are checked for credentials and that by updating this library and in turn this connection variable we are leaving some of those places out.

I've asked the interested parties to test my PR and have heard back from one person so far.